### PR TITLE
Adding support for Azure OpenAI Deployment Types (Global Standard, Standard, Provisioned)

### PR DESCRIPTION
**Pull request by zedhaque**
## Purpose

This PR add support for Azure OpenAI Deployment Types: Global Standard, Standard, and Provisioned. The default deployment type remains Standard.

## Does this introduce a breaking change?

When developers merge from main and run the server, azd up, or azd deploy, will this produce an error?
If you're not sure, try it out on an old environment.

```
[ ] Yes
[X] No
```

## Does this require changes to learn.microsoft.com docs?

This repository is referenced by [this tutorial](https://learn.microsoft.com/azure/developer/python/get-started-app-chat-template)
which includes deployment, settings and usage instructions. If text or screenshot need to change in the tutorial,
check the box below and notify the tutorial author. A Microsoft employee can do this for you if you're an external contributor.

```
[ ] Yes
[X ] No
```

## Type of change

```
[ ] Bugfix
[X ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Code quality checklist

See [CONTRIBUTING.md](https://github.com/Azure-Samples/azure-search-openai-demo/blob/main/CONTRIBUTING.md#submit-pr) for more details.

- [ ] The current tests all pass (`python -m pytest`).
- [ ] I added tests that prove my fix is effective or that my feature works
- [ ] I ran `python -m pytest --cov` to verify 100% coverage of added lines
- [ ] I ran `python -m mypy` to check for type errors
- [ ] I either used the pre-commit hooks or ran `ruff` and `black` manually on my code.

### CODE REVIEW

**Comment by pamelafox** [comment]
Thanks Zed! We actually are hoping to be able to default to GlobalStandard in the future, so this helps set us up for that. What about adding the sku name for the embedding deployment and vision deployment as well? I would think that if you need GlobalStandard for one, you'd need it for the others?

**Comment by pamelafox** [comment]
@zedhaque Yes please make those configurable as well.

File: infra/main.parameters.json [hunk 1]
```diff
@@ -110,6 +113,9 @@
     "embeddingDeploymentVersion":{
       "value": "${AZURE_OPENAI_EMB_DEPLOYMENT_VERSION}"
     },
+    "cembeddingDeploymentSkuName":{
```
Comment (pamelafox) [inline infra/main.parameters.json:116]: Typo here
```suggestion
    "embeddingDeploymentSkuName":{
```

File: infra/main.parameters.json [hunk 2]
```diff
@@ -98,6 +98,9 @@
     "chatGptDeploymentVersion":{
       "value": "${AZURE_OPENAI_CHATGPT_DEPLOYMENT_VERSION}"
     },
+    "chatGptDeploymentSkuName":{
+      "value": "${AZURE_OPENAI_CHATGPT_DEPLOYMENT_SKUNAME}"
```
Comment (pamelafox) [inline infra/main.parameters.json:102]: For similarity with other SKU variables in the JSON

```suggestion
      "value": "${AZURE_OPENAI_CHATGPT_DEPLOYMENT_SKU}"
```

File: infra/main.parameters.json [hunk 3]
```diff
@@ -110,6 +113,9 @@
     "embeddingDeploymentVersion":{
       "value": "${AZURE_OPENAI_EMB_DEPLOYMENT_VERSION}"
     },
+    "cembeddingDeploymentSkuName":{
+      "value": "${AZURE_OPENAI_EMB_DEPLOYMENT_SKUNAME}"
```
Comment (pamelafox) [inline infra/main.parameters.json:117]: Ditto, for similarity with other SKU variables:

```suggestion
      "value": "${AZURE_OPENAI_EMB_DEPLOYMENT_SKU}"
```

File: docs/deploy_features.md [hunk 4]
```diff
@@ -128,7 +135,13 @@ By default, the deployed Azure web app uses the `text-embedding-ada-002` embeddi
     azd env set AZURE_OPENAI_EMB_DEPLOYMENT_VERSION 1
     ```
 
-4. When prompted during `azd up`, make sure to select a region for the OpenAI resource group location that supports the text-embedding-3 models. There are [limited regions available](https://learn.microsoft.com/azure/ai-services/openai/concepts/models#embeddings-models).
+4. Set the deployment Sku name to "GlobalStandard" :
```
Comment (pamelafox) [inline docs/deploy_features.md:138]: I realized that we dont yet have embedding models for GlobalStandard, so you can remove this bullet point for now. Can bring it back once we have those models. Still nice to have support for it in the Bicep though.

File: docs/deploy_features.md [hunk 5]
```diff
@@ -59,6 +59,12 @@ Execute the following commands inside your terminal:
     azd env set AZURE_OPENAI_CHATGPT_MODEL gpt-4o-mini
     ```
 
+1. To set the Azure OpenAI deployment Sku name, run this command with the desired Sku name.
+
+    ```bash
+    azd env set AZURE_OPENAI_CHATGPT_DEPLOYMENT_SKUNAME GlobalStandard
```
Comment (pamelafox) [inline docs/deploy_features.md:65]:
```suggestion
    azd env set AZURE_OPENAI_CHATGPT_DEPLOYMENT_SKU GlobalStandard
```

File: docs/deploy_features.md [hunk 6]
```diff
@@ -59,6 +59,12 @@ Execute the following commands inside your terminal:
     azd env set AZURE_OPENAI_CHATGPT_MODEL gpt-4o-mini
     ```
 
+1. To set the Azure OpenAI deployment Sku name, run this command with the desired Sku name.
```
Comment (pamelafox) [inline docs/deploy_features.md:62]:
```suggestion
1. To set the Azure OpenAI deployment SKU name, run this command with [the desired SKU name](https://learn.microsoft.com/azure/ai-services/openai/how-to/deployment-types#deployment-types).
```

**Comment by pamelafox** [comment]
I did an "azd up" from this branch on an existing environment, and that went fine, so it looks like this doesn't introduce any backwards incompatibility issues. I'll merge once CI completes.
Thanks @zedhaque !

**Comment by alenfrey** [comment]
Is there any documentation on how to deploy a provisioned Model after this merge? Thanks. :)

**Comment by alenfrey** [comment]
Thank you so much, I had overlooked that while researching!

**Pull request merged**